test/models/user_test.rb

```diff
@@ -874,4 +876,24 @@ def editing_self_helper
     user.timezone = ''
     assert user.valid?
   end
+
+  test "changing user password as admin without setting current password" do
+    user = FactoryGirl.create(:user)
+    as_admin do
+      user.password = "newpassword"
+      assert user.save
+    end
+    assert user.valid?
+  end
+
+  test "changing user's own password with incorrect current password" do
```

Comment:
It should probably be a model test though, the logic is in the model, not the controller.